### PR TITLE
feat: Add support for Habana DeepSpeed strategy

### PR DESCRIPTION
**Pull request by kaushikb11**
### DeepSpeed Validated Configurations
The following DeepSpeed configurations have been validated to be fully functioning with Gaudi:

- Distributed Data Parallel (multi-card)
- Zero1
- BF16 precision

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/strategies/hpu_deepspeed.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.hpu_parallel import HPUParallelStrategy
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+
+class HPUDeepSpeedStrategy(HPUParallelStrategy):
```
Comment (jerome-habana) [inline src/pytorch_lightning/strategies/hpu_deepspeed.py:27]: We shouldn't be dependent on DDP based strategy for deep speed

Comment (kaushikb11) [reply src/pytorch_lightning/strategies/hpu_deepspeed.py:27]: @jerome-habana why not?

Comment (jerome-habana) [reply src/pytorch_lightning/strategies/hpu_deepspeed.py:27]: generally DS implements the reductions differently unlike DDP.

Comment (kaushikb11) [reply src/pytorch_lightning/strategies/hpu_deepspeed.py:73]: doing the overwriting
